src/GithubMergeTool/GithubMergeTool.cs

```diff
@@ -97,13 +93,57 @@ public class GithubMergeTool
             var jsonBody = JObject.Parse(await response.Content.ReadAsStringAsync());
             if (jsonBody.Type == JTokenType.Array)
             {
-                // Branch doesn't exist
+                // Source branch doesn't exist
                 return (false, response);
             }
 
             var srcSha = ((JValue)jsonBody["object"]["sha"]).ToObject<string>();
 
-            // Create a branch on the repo
+            var existingPR = JArray.Parse(await prsResponse.Content.ReadAsStringAsync()).FirstOrDefault(pr => (string)pr["title"] == prTitle);
+            if (existingPR != null)
+            {
+                if (updateExistingPr)
+                {
+                    // Get the SHA of the PR branch HEAD
+                    var prSha = ((JValue)existingPR["head"]["sha"]).ToObject<string>();
+                    var existingPrNumber = ((JValue)existingPR["number"]).ToObject<string>();
+
+                    if (prSha != srcSha)
```

Comment:
Does this assume that if the source branch is on a different SHA than the merge branch, there are new commits in the source branch? If you pushed a commit to resolve merge conflicts or test failures then would this code throw that away?

Perhaps if "force" were disabled then we would gracefully fail to update the merge PR in the case that new commits were added to it directly?

Reply:
I'm not sure what we will end up with in the PR by using `reset --mixed`. We can do a `git merge --ff-only` instead, which would be a no-op if there's conflict resolving commit in the merge branch.